src/Logic.hs

```diff
@@ -161,7 +161,10 @@ runAction config = foldFree $ \case
     Git.forcePush sha prBranch >>
     Git.tag' sha newTag >>=
     \case TagFailed _ -> cont . (Left "Please check the logs",) <$> Git.push sha (Git.Branch $ Config.branch config)
-          TagOk tagName -> cont . (Right tagName,) <$> Git.pushAtomic [AsRefSpec tagName, AsRefSpec (sha, Git.Branch $ Config.branch config)]
+          TagOk tagName -> cont . (Right tagName,) <$>
+          -- Deleting tag after atomic pushg is important to maintain one "source of truth", namely
+          -- - the origin
```

Comment:
Ah, so that in case we tagged a tag that already exists on the origin, and that is the reason why the push was rejected, we need to get rid of it locally so next time tagging works again as expected?

Reply:
In a nutshell, yes, but the whole case is a more complex beast. Basically, there are 3 failure scenarios (BTW they're all tested):
1. push gets rejected because of conflict (`pushes tags atomically (rejected push)`)
2. push gets rejected because someone created the same tag (`pushes tags atomically (new tag appears)`)
3. push succeeds but someone tagged in the meantime (`tags version consistently across multiple PRs`)

In 1) you end up with a stale tag pointing to some irrelevant commit (in case of conflict the next integration will be tried and it will rebase on `master` again). This commit won't be reachable from the new tip, so it can later create a conflict - you have to delete it
In 2) you have a tag conflict unless you delete the offending tag
In 3) everything is fine until the next fetch. Fetch by default does not include tags, so you won't know about the new tag and it will create a conflict in the future - so you need `fetch --tags`

So theoretically, in presence of `fetch --tags` you could delete the tag only when the push was rejected. But I opted for deleting it in any case. If you pushed successfully then the next integration will fetch it anyways, but:
- it makes the code simpler,
- if something gets messed up in unforeseen ways we do not risk dealing with stale tags (or at least minimize the risk of it) which someone will have to delete manually